DOC-8667: Release notes for v22.2.14 (#17833)
cockroachdb/docs: src/current/_includes/releases/v22.2/v22.2.14.md

## v22.2.14

Release Date: September 11, 2023

{% include releases/release-downloads-docker-image.md release=include.release %}

<h3 id="v22-2-14-sql-language-changes">SQL language changes</h3>

- Added a new syntax to [`SHOW DEFAULT PRIVILEGES`](../v22.2/show-default-privileges.html). Executing `SHOW DEFAULT PRIVILEGES FOR GRANTEE <grantee>` shows the default privileges that a grantee received. [#108283][#108283]

<h3 id="v22-2-14-operational-changes">Operational changes</h3>

- Nodes are now considered suspect when rejoining a cluster and cannot accept lease transfers for one `server.time_after_store_suspect` window, which defaults to 30 seconds. [#107670][#107670]

> - Added two new [changefeed metrics](../v22.2/monitor-and-debug-changefeeds.html#recommended-changefeed-metrics-to-track): `changefeed.checkpoint_progress`  is similar to `changefeed.max_behind_nanos` but supports [metrics labels](../v22.2/monitor-and-debug-changefeeds.html#using-changefeed-metrics-labels); `changefeed.aggregator_progress metric` tracks the progress of individual aggregators (the lowest timestamp for which all aggregators with the label have emitted all values they're responsible for). [#109745][#109745]

Reviewer: ```suggestion
- Added two new [changefeed metrics](../v22.2/monitor-and-debug-changefeeds.html#recommended-changefeed-metrics-to-track): `changefeed.checkpoint_progress`  is similar to `changefeed.max_behind_nanos` but supports [metrics labels](../v22.2/monitor-and-debug-changefeeds.html#using-changefeed-metrics-labels); `changefeed.aggregator_progress` tracks the progress of individual aggregators (the lowest timestamp for which all aggregators with the label have emitted all values they're responsible for). [#109745][#109745]
```
Typo?

Author: yes, typo. updated

- Previously, CockroachDB could encounter an internal error `unexpected non-zero bytes limit for txnKVStreamer` when evaluating [locality-optimized](../v22.2/cost-based-optimizer.html#locality-optimized-search-in-multi-region-clusters) lookup [joins](../v22.2/joins.html) in case it had to perform the remote regions' lookup. The bug was introduced in [v22.2](../releases/v22.2.html) and is now fixed. A temporary workaround can be accomplished without upgrading by setting the `streamer_enabled` [session variable](../v22.2/set-vars.html): `SET streamer_enabled = false;`. [#108252][#108252]
- Fixed a spurious error `no data source matches prefix` that could occur during [planning for a query](../v22.2/cost-based-optimizer.html) with `DISTINCT ON` and [`ORDER BY ASC NULLS LAST`](../v22.2/order-by.html) or `ORDER BY DESC NULLS FIRST`. [#108302][#108302]

> - Previously, using [`IMPORT INTO`](../v22.2/import-into.html) for `DELIMITED DATA` or MySQL imports would error with `column ... does not exist` if it was importing into a [collated string](../v22.2/collate.html) column. This is now fixed. [#108287][#108287]

Reviewer: ```suggestion
- Fixed a bug where using [`IMPORT INTO`](../v22.2/import-into.html) for `DELIMITED DATA` or MySQL imports would error with `column ... does not exist` if it was importing into a [collated string](../v22.2/collate.html) column. [#108287][#108287]
```

Author: updated.

> - Fixed a bug in the index recommendations provided in the EXPLAIN output where [`ALTER INDEX ... VISIBLE`](../v22.2/alter-index.html#not-visible) index recommendations may suggest making the wrong index visible when there are multiple invisible indexes in a table. [#108647][#108647]

Reviewer: ```suggestion
- Fixed a bug in the index recommendations provided in the `EXPLAIN` output where [`ALTER INDEX ... VISIBLE`](../v22.2/alter-index.html#not-visible) index recommendations may suggest making the wrong index visible when there are multiple invisible indexes in a table. [#108647][#108647]
```

Author: formatted as code and added a link for `EXPLAIN`.

> - Fixed a bug that could cause a query with [`LIMIT`](../v22.2/limit-offset.html#limit) and [`ORDER BY`](../v22.2/order-by.html) to return results in the wrong order. This bug could cause incorrect results as well if the `LIMIT` was nested within an outer query, for example, under another `LIMIT`. The bug existed prior to v22.2. [#107379][#107379]

Reviewer: ```suggestion
- Fixed a bug that could cause a query with [`LIMIT`](../v22.2/limit-offset.html#limit) and [`ORDER BY`](../v22.2/order-by.html) to return results in the wrong order. This bug could also cause incorrect results if the `LIMIT` was nested within an outer query, for example, under another `LIMIT`. [#107379][#107379]
```

Author: updated

> - Fixed a bug introduced in [v22.1](../releases/v22.1.html) that could cause a [join](../v22.2/joins.html) to infinite-loop in rare cases when (1) the join filter is not an equality and (2) no columns from the left input are returned. [#106874][#106874]

Reviewer: ```suggestion
- Fixed a bug introduced in [v22.1](../releases/v22.1.html) that could cause a [join](../v22.2/joins.html) to infinite-loop in rare cases when the join filter is not an equality and no columns from the left input are returned. [#106874][#106874]
```

Author: updated
